docs/core/tools/dotnet-build.md

```diff
@@ -4,54 +4,51 @@ description: The dotnet-build command builds a project and all of its dependenci
+The `dotnet build` command builds the project and its dependencies into a set of binaries. The binaries include the project's code in Intermediate Language (IL) files with a `*.dll` extension and symbol files used for debugging with a `*.pdb` extension. A dependencies JSON file (`*.deps.json`) is produced that lists the dependencies of the application. A `runtime.config.json` file is produced, which specifies the shared runtime and its version for the application.
 
-Building requires the existence of an *assets.json* file (a file that lists all of the dependencies of your application), which means that you have to run [`dotnet restore`](dotnet-restore.md) prior to building the project. Lack of the assets file manifests as the inability of the tooling to resolve reference assemblies which will result in errors. 
+If the project has third-party dependencies, such as libraries from NuGet, they're resolved from the NuGet cache and aren't available with the project's built output. With that in mind, the product of `dotnet build` isn't ready to be transferred to another machine to run. This is in contrast to the behavior of the .NET Framework in which building an executable project (an application) produces output that's runnable on any machine where the .NET Framework is installed. To have a similar experience with .NET Core, you use the [dotnet publish](dotnet-publish.md) command. For more information, see [.NET Core Application Deployment](../deploying/index.md). 
 
-`dotnet build` uses MSBuild to build the project, thus it supports both parallel builds and incremental builds. Please refer to [MSBuild documentation](https://docs.microsoft.com/visualstudio/msbuild/msbuild) to get more information on those topics. 
+Building requires the *project.assets.json* file, which lists the dependencies of your application. The file is created when you execute [`dotnet restore`](dotnet-restore.md) before building the project. Without the assets file in place, the tooling cannot resolve reference assemblies, which will result in errors.
 
-In addition to its options, the `dotnet build` command will accept MSBuild options as well, such as `/p` for setting properties or `/l` to define a logger. You can find out more about these options in the [`dotnet msbuild`](dotnet-msbuild.md) command documentation. 
+`dotnet build` uses MSBuild to build the project; thus, it supports both parallel and incremental builds. Refer to [MSBuild documentation](https://docs.microsoft.com/visualstudio/msbuild/msbuild) for more information. 
```

Comment:
for incremental, I'm wondering if we should link directly to the topic: https://docs.microsoft.com/visualstudio/msbuild/incremental-builds